src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -499,8 +513,19 @@ private Task ProcessStreamItem(HubConnectionContext connection, StreamItemMessag
             Log.StreamingResult(_logger, invocationId, descriptor.MethodExecutor);
             var streamItemMessage = new StreamItemMessage(invocationId, null);
 
+            var count = 0;
             while (await enumerator.MoveNextAsync())
             {
+                if (activity is not null)
+                {
+                    count++;
+                    // https://github.com/open-telemetry/semantic-conventions/blob/main/docs/rpc/rpc-spans.md#events
+                    var tags = new ActivityTagsCollection(
+                        [new("rpc.message.type", "SENT"), new("rpc.message.id", count)]);
+                    var @event = new ActivityEvent("rpc.message", tags: tags);
+                    activity.AddEvent(@event);
```

Comment:
Isn't there a memory concern here? Can this OOM eventually? What if I have a long stream.

Reply:
I thought someone mentioned it already having a limit. But looking at `AddEvent` I don't see any limit. So yes, we should add one.

Reply:
I feel like we should remove this wholesale.

Reply:
There is some debate in the OTel community about whether we should use events on traces or log messages. Now that traces and logs can be correlated, using logs is good. Maybe this is a place that should be considered? It also means you can set a lowish level and that can be configured on a per-provider basis.

Reply:
I'm not sure those events are super-helpful even if we eliminate memory concern. I remember playing with it and it was really hard to correlate these events or make sense what's going on based on them.

I'd expect users who care about the details to create custom activities on client/server for small operations. And events would be a confusing noise for those who don't care. 

I'd just rely on the existing logs (I assume there are some already) for now. OTel will eventually unify span events into logs.

Reply:
+1

Let's remove events. Nothing prevents us adding new events in the future.

Reply:
I would suggest changing them to logs, and we can then pick an appropriate verbosity level.

Reply:
We should remove anything from this code path. We already have verbose logs at other layers.

Reply:
Not sure, but I don't think it's important enough to log every stream item. Though I'd be interested to know if we did something similar in grpc.